### Import zbioru danych

In [ ]:
import pandas as pd

df = pd.read_csv('AmesHousing.csv')

### Rozmiar zbioru danych

In [24]:
df.shape

(2930, 82)

Nasz `unikalny` zbior danych sklada sie z `2930` obserwacji oraz `82` cech.

### Unikatowe Wartości

In [38]:
uniqueCounts = df.nunique().sort_values(ascending=True)

uniqueCounts_dataFrame = uniqueCounts.reset_index()
uniqueCounts_dataFrame.columns = ['Feature', 'Unique Values']
uniqueCounts_dataFrame
# print(uniqueCounts_dataFrame.to_string(index=False))

,Feature,Unique Values
0,Central Air,2
1,Street,2
2,Alley,2
3,Bsmt Half Bath,3
4,Paved Drive,3
...,...,...
77,Bsmt Unf SF,1137
78,Gr Liv Area,1292
79,Lot Area,1960
80,PID,2930


Powyższa tabela składa się z trzech kolumn:

1. **Index**  
   Automatycznie wygenerowany numer wiersza

2. **Feature**  
   Nazwa cechy (kolumny) w zbiorze danych.

3. **Unique Values**  
   Liczba unikatowych wartości przyjmowanych przez daną cechę w całym zbiorze obserwacji.  
---
- Jeśli **Unique Values = 2**, to zmienna jest binarna (np. 0/1 lub TAK/NIE).  
  – Przykład: `Central Air` przyjmuje wartości `Y`/`N`, stąd ma 2 unikalne wartości.

- **Cechy dyskretne** to te o niewielkiej liczbie unikatów (zwykle 3–10).  
  – Przykłady: `Fireplace` przyjmuje juz 5 wartosci

- **Cechy ciągłe** mają bardzo dużo unikatowych wartości,   
  – Przykłady: `Lot Area` przyjmuje 1960 unikalnych wartosci

Dzięki temu podziałowi łatwo zidentyfikujemy, które cechy traktujemy jako dyskretne, a które jako ciągłe i z nich wybierzemy dwa zestawy po 7 cech dla kazdego typu


### Selekcja cech dyskretnych oraz ciągłych

In [36]:
discreteFeatures = [
    'Central Air',
    'Paved Drive',
    'Bsmt Full Bath',
    'Kitchen AbvGr',
    'Fireplaces',
    'Bsmt Half Bath',
    'Garage Cars'
]

1. **`Central Air`**  
   - **Opis:** Czy budynek posiada klimatyzację (`Y`/`N`).  
   - **Typ:** Zmienna binarna (2 wartości).  
   - **Zakres wartości:** `Y` (tak), `N` (nie).  

2. **`Paved Drive`**  
   - **Opis:** Rodzaj utwardzenia podjazdu.  
   - **Typ:** Zmienna nominalna.  
   - **Zakres wartości:** `N` (brak utwardzenia), `P` (częściowe), `Y` (tak).  

3. **`Bsmt Full Bath`**  
   - **Opis:** Liczba pełnych łazienek w piwnicy.  
   - **Typ:** Zmienna licznikowa.  
   - **Zakres wartości:** 0–3.  

4. **`Kitchen AbvGr`**  
   - **Opis:** Liczba kuchni nad poziomem gruntu.  
   - **Typ:** Zmienna licznikowa.  
   - **Zakres wartości:** zwykle 1–2.  

5. **`Fireplaces`**  
   - **Opis:** Liczba kominków w domu.  
   - **Typ:** Zmienna licznikowa.  
   - **Zakres wartości:** 0–4.  

6. **`Bsmt Half Bath`**  
   - **Opis:** Liczba „połówkowych” łazienek w piwnicy.  
   - **Typ:** Zmienna licznikowa.  
   - **Zakres wartości:** 0–2.  

7. **`Garage Cars`**  
   - **Opis:** Liczba miejsc parkingowych w garażu.  
   - **Typ:** Zmienna licznikowa.  
   - **Zakres wartości:** 0–4.  

---

*Zdecydowalismy sie na wybor takich cech, poniewaz kazda z nich ma liczbe unikatowych wartosci (<10), dobrze bedzie sie je wizualizowac, uwazamy ze dobrze oddadza wplyw na atrakcyjnosc mieszkania*


In [ ]:
continuousFeatures = [
    'Lot Area',
    'Gr Liv Area',
    'Total Bsmt SF',
    '1st Flr SF',
    '2nd Flr SF',
    'Garage Area',
    'SalePrice'
]

1. **`Lot Area`**  
   - **Opis:** Powierzchnia działki w stopach kwadratowych.  
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od kilku tysięcy do ponad 200 000 ft².

2. **`Gr Liv Area`**  
   - **Opis:** Powierzchnia użytkowa mieszkalna powyżej poziomu gruntu w stopach kwadratowych. 
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od kilkuset do ponad 4500 ft².

3. **`Total Bsmt SF`**  
   - **Opis:** Całkowita powierzchnia piwnicy w stopach kwadratowych.  
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od 0 (brak piwnicy) do około 6000 w stopach kwadratowych.

4. **`1st Flr SF`**  
   - **Opis:** Powierzchnia parteru w stopach kwadratowych.  
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od kilkuset do ponad 4000 ft².

5. **`2nd Flr SF`**  
   - **Opis:** Powierzchnia piętra w stopach kwadratowych. 
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od 0 (domy parterowe) do około 2000 ft².

6. **`Garage Area`**  
   - **Opis:** Powierzchnia garażu w stopach kwadratowych. 
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od 0 (brak garażu) do ponad 1400 ft².

7. **`SalePrice`**  
   - **Opis:** Cena sprzedaży domu w dolarach amerykańskich.  
   - **Typ:** Zmienna ciągła.  
   - **Zakres wartości:** od kilkudziesięciu tysięcy do ponad miliona dolarów.


In [42]:
myData = discreteFeatures + continuousFeatures
summary = pd.DataFrame({
    'Feature': myData,
    'Type':   ['discrete']*len(discreteFeatures) + ['continuous']*len(continuousFeatures),
    'Unique Values': [df[f].nunique() for f in myData]
})
summary

,Feature,Type,Unique Values
0,Central Air,discrete,2
1,Paved Drive,discrete,3
2,Bsmt Full Bath,discrete,4
3,Kitchen AbvGr,discrete,4
4,Fireplaces,discrete,5
5,Bsmt Half Bath,discrete,3
6,Garage Cars,discrete,6
7,Lot Area,continuous,1960
8,Gr Liv Area,continuous,1292
9,Total Bsmt SF,continuous,1058


Nasz nowy zbiór, na którym będziemy pracować, składa się z:
- **14 cech**:  
  - 7 cech dyskretnych  
  - 7 cech ciągłych  
- **2930 obserwacji**  

Ograniczylismy sie do tych 14 cech, zatem duzo latwiejsze staje sie przeprowadzenie analizy dancyh jak i sama wizualizacja danych.  
Dodatkwo zachowujemy balans miedzy zmiennyli liczbowymi a kategorycznymi.


### Opis zbioru danych

TODO:
- zbior z kaggle
- opisac zbior description z kaggle
- liczba obserwacji liczba cech

- wykresy do wizulaizacji tych cech czyli pewnie gdzies z 14 wykresow dla kazdej cechy jej rozklad bo jest 3000 obserwacji
